app/models/test_run.rb

```diff
@@ -237,6 +237,11 @@ def trigger_build!
     Houston.observer.fire "test_run:start", test_run: self
   end
 
+  def trigger_retry!
+    project.ci_server.respond_to?(:retry!) ? project.ci_server.retry!(sha) : project.ci_server.build!(sha)
```

Comment:
Or instead of checking `respond_to?(:retry!)` maybe just have the Jenkins adapter alias `build!` as `retry!`?

Per your commit description, it seems reasonable to make this part of the public API for a CI Server adapter

Reply:
Makes sense. I'm good with making that change a part of this PR 👍